src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/contact/ContactRepository.kt

```diff
@@ -38,4 +41,19 @@ data class Contact(
   val created: Instant,
 
   val updated: Instant
-)
+) {
+  override fun equals(other: Any?): Boolean {
+    if (this === other) return true
+    if (other == null || Hibernate.getClass(this) != Hibernate.getClass(other)) return false
+    other as Contact
+
+    return id != null && id == other.id
+  }
```

Comment:
Generated by JPA Buddy which should avoid some pitfalls using Kotlin data classes and JPA. A side effect is that you cannot use equals on unsaved entities, but this is currently only needed for tests so we work around it.